lib/ansible/module_utils/vmware.py

```diff
@@ -887,12 +887,13 @@ def get_vm(self):
         vm_obj = None
         user_desired_path = None
         use_instance_uuid = self.params.get('use_instance_uuid') or False
-        if self.params['uuid'] and not use_instance_uuid:
-            vm_obj = find_vm_by_id(self.content, vm_id=self.params['uuid'], vm_id_type="uuid")
-        elif self.params['uuid'] and use_instance_uuid:
-            vm_obj = find_vm_by_id(self.content,
-                                   vm_id=self.params['uuid'],
-                                   vm_id_type="instance_uuid")
+        if 'uuid' in self.params:
```

Comment:
I would suggest to create a separate PR for this change. Since it would be easy to backport this to Stable version.